scripts/us_census/acs5yr/subject_tables/common/generateColMap.py

```diff
@@ -0,0 +1,94 @@
+    self.columnList = columnList
+    self.columnMap = {}
+    self.makeColMap()
+
+  def makeColMap(self):
+    #for each column generate the definition of their respective statistical variable node
+    for col in self.columnList:
+      if 'ignoreColumns' in self.features and col not in self.features['ignoreColumns']:
+          self.columnMap[col] = self.column_to_statVar(col)
+    #TODO: Enable validating the generate columnMap
+    #self.valid_colMap():
+    return self.columnMap
+
+  def column_to_statVar(self, column):
+    """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+    base = False
+    statVar = {}
+    partList = column.split('!!')
+
+    for part in partList:
+      #set the base for special cases like median, etc.
+      if not base and 'measurement' in self.features:
+        if part in self.features['measurement']:
+          statVar.update(self.features['measurement'][part])
+          base = True
+
+    #set the default statVar definition
+    if not base and 'measurement' in self.features and '_DEFAULT' in self.features['measurement']:
+      statVar.update(self.features['measurement']['_DEFAULT'])
+
+    #set the populationType attribute
+    statVar['populationType'] = self.get_populationType(partList)
+
+    # associate pvs to statVar
+    if partList[-1] not in self.features['measurement']:
+      if 'pvs' in self.features:
+        #p = property, k = dictionary of partList[-1] to propertyValue
+        for p, k in self.features['pvs'].items():
+          for c, v in k.items():
+            if c == partList[-1]: #last token of column name matches in property
+              statVar[p] = v
+              if 'inferredSpec' in self.features and p in self.features['inferredSpec']:
+                statVar.update(self.features['inferredSpec'][p])
+            ##if the property is somewhere in the column name
+            for part in partList:
+              if c in part:
+                statVar[p] = v
+
+    #TODO: Add units and scalingFactor
+
+    #if the column is Margin of Error, update statType key
+    if 'Margin of Error' in partList:
+      statVar['statType'] = 'marginOfError'
+
+    #generating dcid using the utils/statvar_dcid_generator.py
+    statVar['dcid'] = dcid_gen.get_stat_var_name(statVar, ignore_props=None)
+
+    ##Add Universe PVs based on the populationType of StatVar
+    if 'universePVs' in self.features:
+      for elem in self.features['universePVs']:
+        if statVar['populationType'] == elem['populationType']:
+          for k, v in elem['dependentPVs'].items():
+            statVar[k] = v
+    return statVar
+  
+  def get_populationType(self, partList):
+    """From tokenized column name, find the most relevant populationType from the JSON Spec """
+    if 'populationType' in self.features:
+      for part in partList:
+        for k, v in self.features['populationType'].items():
+          if k in part:
```

Comment:
why in and not equality (modulo maybe post casing/spacing adjustments)?

Reply:
Reason for using `in` instead of `==`: 
Consider the following column name after splitting on ('!!'): 
```[python]
['Estimate', 'Total Uninsured', 'HOUSEHOLD INCOME (IN 2019 INFLATION ADJUSTED DOLLARS)', 'Total household population', 'Median household income of householders']
```
we want to match for either 'HOUSEHOLD' or 'household' and for both cases, we end up with a case where the expected term to match is a substring. For instance 'HOUSEHOLD' is a substring in 'HOUSEHOLD INCOME (IN 2019 INFLATION ADJUSTED DOLLARS)', and thus we check for the condition using `in`

Reply:
Perhaps we can use `Total household population` as the key? 
And we will expect the spec has the correct column component the describes the populationType. Will that work?